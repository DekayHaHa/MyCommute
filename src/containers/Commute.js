import React from 'react';

export const Commute = () => {
  return (
    <div>

    </div>
  )
}