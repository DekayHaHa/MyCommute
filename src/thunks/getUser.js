import { signinUser, hasError } from '../actions';
import { getUserPreferences } from './getUserPreferences';
import { getUserCommutes } from './getUserCommutes'
import { getUserWeather } from './getUserWeather'

export const getUser = (name, pass) => {
  return async (dispatch) => {
    const optObj = {
      method: 'POST',
      body: JSON.stringify({ name, pass }),
      headers: { "Content-Type": "application/json" }
    }
    try {
      // isLoading
      const response = await fetch('http://localhost:3001/api/user', optObj);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const data = await response.json();
      await dispatch(signinUser(data));
      await dispatch(getUserPreferences(data.id))
      await dispatch(getUserWeather(data.id))
      await dispatch(getUserCommutes(data.id))
      // isLoading
    } catch (error) {
      console.log(error)
      dispatch(hasError(error.message));
    }
  }
}