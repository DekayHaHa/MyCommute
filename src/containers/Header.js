import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import { Link, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getUser } from '../thunks/getUser'
import UserNavBar from './UserNavBar'

export class Login extends Component {


  render() {
    const { user } = this.props
    return (
      <div className='header'>
        <h1 className='title' >MyCommute</h1>
        <Link to='/login'>LOGIN USER</Link>
        <Route path={`/user/${user.id}`} component={UserNavBar} />
      </div>
    );
  }
}

Login.propTypes = {
  user: PropTypes.object,
  signinUser: PropTypes.func
};

export const mapDispatchToProps = (dispatch) => ({
  getUser: (name, pass) => dispatch(getUser(name, pass))
});

export const mapStateToProps = store => ({
  user: store.user,
  error: store.error
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);