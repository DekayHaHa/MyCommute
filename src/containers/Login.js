import React, { Component } from 'react';
// import { Route, Switch } from 'react-router-dom';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getUser } from '../thunks/getUser'

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      userName: '',
      password: ''
    }
  }
  handleSubmit = (e) => {
    e.preventDefault()
    const { userName, password } = this.state
    this.props.getUser(userName, password)
    this.setState({ userName: '', password: '' })
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  render() {
    const { user, error } = this.props
    const { userName, password } = this.state
    return (
      <div>
        {user.id && <Redirect to={`/user/${user.id}/commutes`} />}
        <form onSubmit={this.handleSubmit}>
          <input className='input-button'
            name='userName'
            type='text'
            value={userName}
            placeholder='USERNAME...'
            onChange={this.handleChange}
          />
          <input className='input-button'
            name='password'
            type='password'
            value={password}
            placeholder='PASSWORD...'
            onChange={this.handleChange}
          />
          <button className='form-button' onClick={this.fetchData}>Sign In</button>
          {error && <p className='input-error' >{error}</p>}
        </form>
      </div>
    );
  }
}

Login.propTypes = {
  user: PropTypes.object,
  getUser: PropTypes.func,
  error: PropTypes.string
};

export const mapDispatchToProps = (dispatch) => ({
  getUser: (name, pass) => dispatch(getUser(name, pass))
});

export const mapStateToProps = store => ({
  user: store.user,
  error: store.error
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);