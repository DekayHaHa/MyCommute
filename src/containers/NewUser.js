import React, { Component } from 'react';
// import { Route, Switch } from 'react-router-dom';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createUser } from '../thunks/createUser'

export class NewUser extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      userName: '',
      password: '',
      verifyPassword: '',
      zip: 0,
      lat: 0,
      lng: 0,
      locCheck: false,
    }
  }

  componentDidMount() {
    let answer = window.confirm('Can we use your web browesr location?')
    if (answer) {
      this.setState({ locCheck: true })
    }
  }

  handleSubmit = (e) => {
    e.preventDefault()
    console.log(this.state)
    const { userName, password, zip, lat, lng, email } = this.state
    let user = {
      email,
      userName,
      password,
      zip,
      lat,
      lng
    }
    this.props.createUser(user)
    this.setState({ userName: '', password: '', verifyPassword: '', email: '', zip: 0, lat: 0, lng: 0 })
  }

  successCallback = (pos) => {
    this.setState({ lat: pos.coords.latitude, lng: pos.coords.longitude })
  }

  errorCallback = (err) => {
    console.log('in err', err)
    return err
  }

  getUserLocation = () => {
    const opt = { timeout: 10000 }
    navigator.geolocation.getCurrentPosition(this.successCallback, this.errorCallback, opt);
    console.log(this.state)
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  render() {
    const { user, error } = this.props
    const { userName, password, verifyPassword, email, zip, lat, lng, locCheck } = this.state
    if (!lat && !lng && locCheck) this.getUserLocation()
    return (
      <div>
        {user.id && <Redirect to='/' />}
        <form onSubmit={this.handleSubmit}>
          <input
            name='email'
            type='text'
            value={email}
            placeholder='EMAIL...'
            onChange={this.handleChange}
          />
          <input
            name='userName'
            type='text'
            value={userName}
            placeholder='USERNAME...'
            onChange={this.handleChange}
          />
          <input
            name='zip'
            type='number'
            value={zip}
            placeholder='ZIP CODE...'
            onChange={this.handleChange}
          />
          {zip.length > 0 && zip.length !== 5 && <p>Please enter a 5 digit Zip Code.</p>}
          <input
            name='password'
            type='password'
            value={password}
            placeholder='PASSWORD...'
            onChange={this.handleChange}
          />
          <input
            name='verifyPassword'
            type='password'
            value={verifyPassword}
            placeholder='VERIFY PASSWORD...'
            onChange={this.handleChange}
          />
          {verifyPassword !== password && <p>Passwords Do Not Match</p>}
          {error && <p>{error}</p>}
          <button onClick={this.fetchData}>Sign Up</button>
        </form>
      </div>
    );
  }
}

NewUser.propTypes = {
  user: PropTypes.object,
  createUser: PropTypes.func,
  error: PropTypes.string
};

export const mapDispatchToProps = (dispatch) => ({
  createUser: (user) => dispatch(createUser(user))
});

export const mapStateToProps = store => ({
  user: store.user,
  error: store.error
});

export default connect(mapStateToProps, mapDispatchToProps)(NewUser);