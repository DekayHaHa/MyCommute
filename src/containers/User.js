import React, { Component } from 'react';
// import { Route, Switch } from 'react-router-dom';
import { Link, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getUser } from '../thunks/getUser'

export class User extends Component {


  render() {
    return (
      <nav>

      </nav>
    );
  }
}

User.propTypes = {
  user: PropTypes.object,
  signinUser: PropTypes.func
};

export const mapDispatchToProps = (dispatch) => ({
  getUser: (name, pass) => dispatch(getUser(name, pass))
});

export const mapStateToProps = store => ({
  user: store.user,
  error: store.error
});

export default connect(mapStateToProps, mapDispatchToProps)(User);