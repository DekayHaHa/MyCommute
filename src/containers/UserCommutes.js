import React, { Component } from 'react';
// import { Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getUser } from '../thunks/getUser'
import moment from 'moment'

export class UserCommutes extends Component {

  checkPref = () => {
    const { pref } = this.props
    // const timeDiff = ( - Date.now())
    console.log(moment(pref.morning, "LT").format())
  }

  render() {

    return (
      <div>
        <h3>I'm Commutes</h3>
        <p>Moring commute {this.checkPref()}</p>
      </div>
    );
  }
}

UserCommutes.propTypes = {
  user: PropTypes.object,
  signinUser: PropTypes.func
};

export const mapDispatchToProps = (dispatch) => ({
  getUser: (name, pass) => dispatch(getUser(name, pass))
});

export const mapStateToProps = store => ({
  user: store.user,
  error: store.error,
  pref: store.pref,
  commutes: store.commutes
});

export default connect(mapStateToProps, mapDispatchToProps)(UserCommutes);