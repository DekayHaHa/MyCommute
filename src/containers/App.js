import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import Login from './Login'
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Header from './Header'
// import NewUser from './NewUser'
import UserCommutes from './UserCommutes'
import UserWeather from './UserWeather'
import UserPreferences from './UserPreferences'
import { Home } from '../components/Home'
import { PageError } from '../components/PageError'
import '../styles/App.css'

export class App extends Component {

  render() {
    const { user } = this.props
    return (
      <div className="App">
        {/* {user.id && this.getUserWeather()} */}
        <Route to='/' component={Header} />
        <Switch>
          <Route exact path='/' component={Home} />
          <Route exact path='/login' component={Login} />
          <Route exact path={`/user/${user.id}/commutes`} component={UserCommutes} />
          <Route exact path={`/user/${user.id}/preferences`} component={UserPreferences} />
          <Route exact path={`/user/${user.id}/weather`} component={UserWeather} />
          {/* <Route exact path='/newUser' component={NewUser} /> */}
          <Route component={PageError} />
        </Switch>
      </div>
    );
  }
}

App.propTypes = {
  user: PropTypes.object
};

export const mapStateToProps = store => ({
  user: store.user
});

export default connect(mapStateToProps)(App);