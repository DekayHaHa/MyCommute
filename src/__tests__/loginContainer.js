import React from 'react'
import { Login, mapDispatchToProps, mapStateToProps } from '../containers/Login'
import { shallow } from 'enzyme'
import { getUser } from '../thunks/getUser'

jest.mock('../thunks/getUser')

describe('Login', () => {
  let wrapper;
  const mockUser = { id: 1, userName: 'mahk' }
  beforeEach(() => {
    wrapper = shallow(<Login user={mockUser} />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('map props', () => {
    it('should map state to props', () => {
      const mockState = {
        user: { id: 1, userName: 'mahk' },
        error: 'im wrong',
        pref: { id: 1 },
        commutes: { id: 1 },
        weather: { current: {} },
        notMe: 'shouldnt show'
      }
      const expectedState = {
        user: { id: 1, userName: 'mahk' },
        error: 'im wrong',
      }
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedState);
    })

    it('should map dispatch to props', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getUser()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.getUser()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})