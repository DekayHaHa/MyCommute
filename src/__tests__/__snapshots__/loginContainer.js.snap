// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    user={
      Object {
        "id": 1,
        "userName": "mahk",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Redirect
          to="/user/1/commutes"
        />,
        <form
          onSubmit={[Function]}
        >
          <input
            className="input-button"
            name="userName"
            onChange={[Function]}
            placeholder="USERNAME..."
            type="text"
            value=""
          />
          <input
            className="input-button"
            name="password"
            onChange={[Function]}
            placeholder="PASSWORD..."
            type="password"
            value=""
          />
          <button
            className="form-button"
          >
            Sign In
          </button>
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "to": "/user/1/commutes",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="input-button"
              name="userName"
              onChange={[Function]}
              placeholder="USERNAME..."
              type="text"
              value=""
            />,
            <input
              className="input-button"
              name="password"
              onChange={[Function]}
              placeholder="PASSWORD..."
              type="password"
              value=""
            />,
            <button
              className="form-button"
            >
              Sign In
            </button>,
            undefined,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-button",
              "name": "userName",
              "onChange": [Function],
              "placeholder": "USERNAME...",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-button",
              "name": "password",
              "onChange": [Function],
              "placeholder": "PASSWORD...",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "className": "form-button",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
          undefined,
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Redirect
            to="/user/1/commutes"
          />,
          <form
            onSubmit={[Function]}
          >
            <input
              className="input-button"
              name="userName"
              onChange={[Function]}
              placeholder="USERNAME..."
              type="text"
              value=""
            />
            <input
              className="input-button"
              name="password"
              onChange={[Function]}
              placeholder="PASSWORD..."
              type="password"
              value=""
            />
            <button
              className="form-button"
            >
              Sign In
            </button>
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "to": "/user/1/commutes",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="input-button"
                name="userName"
                onChange={[Function]}
                placeholder="USERNAME..."
                type="text"
                value=""
              />,
              <input
                className="input-button"
                name="password"
                onChange={[Function]}
                placeholder="PASSWORD..."
                type="password"
                value=""
              />,
              <button
                className="form-button"
              >
                Sign In
              </button>,
              undefined,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-button",
                "name": "userName",
                "onChange": [Function],
                "placeholder": "USERNAME...",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-button",
                "name": "password",
                "onChange": [Function],
                "placeholder": "PASSWORD...",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "className": "form-button",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "button",
            },
            undefined,
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
